packages/auth/amplify_auth_cognito_dart/lib/src/model/session/cognito_auth_session_result.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_auth_cognito.model.cognito_auth_session_result}
+/// Wraps a session value.
+/// {@endtemplate}
+///
+/// The result may contain an error even though the method
+/// fetching it had a success callback.
+class CognitoAuthSessionResult<T>
+    with AWSEquatable<CognitoAuthSessionResult<T>> {
```

Comment:
This can be AWSDebuggable, too - only values which are also AWSDebuggable will print, so it's safe